File: electron/src/protocol/Request.ts
/**
 * Created by John on 31-05-2016.
 */
export class Request {
    static messageType: string = "REQUEST";
    static RUN_MODEL: string = "RunModel";
    static GET_MODEL_INFO: string = "GetModelInfo";
    static GET_CLASS_INFO: string = "GetClassInfo";
    static GET_FUNCTION_INFO: string = "GetFunctionInfo";
    static SET_ROOT_CLASS: string = "SetRootClass";
    static STOP_SERVER: string = "StopServer";
    request: string ;
    parameter: string ;
}

File: core/interpreter-json-server/src/main/java/org/overturetool/plotting/protocol/Request.java
package org.overturetool.plotting.protocol;

public class Request
{
	public static final String messageType = "REQUEST";
	public static final String RUN_MODEL = "RunModel";
	public static final String GET_MODEL_INFO = "GetModelInfo";
	public static final String GET_CLASS_INFO = "GetClassInfo";
	public static final String GET_FUNCTION_INFO = "GetFunctionInfo";
	public static final String SET_ROOT_CLASS = "SetRootClass";
	public static final String STOP_SERVER = "StopServer";
	public String request;
	public String parameter;

	public Request(String request, String parameter)
	{
		this.request = request;
		this.parameter = parameter;
	}
	public Request(String request)
	{
		this.request = request;
	}

	public Request()
	{

	}
}
